core/async_cell.rs

```diff
@@ -45,6 +51,15 @@ impl<T: 'static> AsyncRefCell<T> {
   pub fn as_ptr(&self) -> *mut T {
     self.value.get()
   }
+
+  /// Returns value of this cell if there are no outstanding borrows.
+  pub fn try_unwrap(self) -> Result<T, AsyncRefCell<T>> {
+    if self.borrow_count.get().is_empty() {
+      Ok(self.value.into_inner())
+    } else {
```

Comment:
* Is this possible? It seems that since this method consumes self there cannot be any outstanding borrows.
* Canonically this method should be called "into inner" I think.

Reply:
Ah good catch, you're right, first we need to call `Rc::into_inner` which gives ownership to the `AsyncRefCell` so checking borrow count is not necessary. I'll rename to `into_inner`